### DISCUSSION TITLE
HOW TO ADD CAMERA FOLLOWS CHARACTER ON NOTE HIT EFFECT IN YOSHI ENGINE

### DISCUSSION THREAD

**Discussion by Mlaofmd** [YoshiCrafter29/YoshiCrafterEngine#346] [Q&A] [answered]
hello i need add camera follows character on note hit how like sonic.exe and wednesday infidelity

**Comment by ghost** [comment, marked as answer]
Go to your global_scripts folder (if you don't have one in your mods folder, create one), make a new hx file and name the file something like this: camFollow.hx then open the file and paste this code and save it:

import("Reflect");
import("Main");
import("StringTools");
var engineVer = Reflect.getProperty(Main, "engineVer");
if (!StringTools.startsWith(engineVer, "1.")) {
  function createPost() {
    for(char in [PlayState.boyfriend, PlayState.dad]) {
      char.addCameraOffset('singLEFT', -50, 0);
      char.addCameraOffset('singRIGHT', 50, 0);
      char.addCameraOffset('singUP', 0, -50);
      char.addCameraOffset('singDOWN', 0, 50);
    }
  }
} else {
  trace("your yoshicrafter engine is outdated bro, you should consider updating");
  function postUpdate(elapsed:Float) {
    if (PlayState.section == null) return;
    var animName = "";
    if (PlayState.section.mustHitSection)
        animName = PlayState.boyfriend.animation.curAnim.name;
    else
        animName = PlayState.dad.animation.curAnim.name;
    if (animName == 'singLEFT') {
      PlayState.camFollow.x -= 50;
    }
    if (animName == 'singRIGHT') {
      PlayState.camFollow.x += 50;
    }
    if (animName == 'singUP') {
      PlayState.camFollow.y -= 50;
    }
    if (animName == 'singDOWN') {
      PlayState.camFollow.y += 50;
    }
  }
}

**Reply by Mlaofmd** [reply]
TY!!!!!!
